nodebuilder/rpc_test.go

```diff
@@ -0,0 +1,121 @@
+package nodebuilder
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/api/mocks"
+	"github.com/celestiaorg/celestia-node/api/rpc"
+	"github.com/celestiaorg/celestia-node/api/rpc/client"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+func TestRPCCallsUnderlyingNode(t *testing.T) {
+	nd, server := setupNodeWithModifiedRPC(t)
+	url := nd.RPCServer.ListenAddr()
+	client, closer, err := client.NewClient(context.Background(), "http://"+url)
+	require.NoError(t, err)
+	defer closer.CloseAll()
+	ctx := context.Background()
+
+	expectedBalance := &state.Balance{
+		Amount: sdk.NewInt(100),
+		Denom:  "utia",
+	}
+
+	server.EXPECT().Balance(gomock.Any()).Return(expectedBalance, nil).Times(1)
```

Comment:
It's not worth having huge API mock just for a one line if we don't plan to add more similar tests